bundles/org.eclipse.text.quicksearch/src/org/eclipse/text/quicksearch/internal/ui/QuickSearchDialog.java

```diff
@@ -409,16 +412,21 @@ public void create() {
 	 */
 	protected void restoreDialog(IDialogSettings settings) {
 		try {
-			if (initialPatternText==null) {
-				String lastSearch = settings.get(DIALOG_LAST_QUERY);
+			DialogSettings dialogSettings = (DialogSettings) settings;
+
+			if (initialPatternText==null && dialogSettings.getLastFilters() != null) {
+				String lastSearch = dialogSettings.getLastFilters().getFirst();
+
 				if (lastSearch==null) {
 					lastSearch = EMPTY_STRING;
 				}
 				pattern.setText(lastSearch);
 				pattern.selectAll();
 			}
-			if (settings.get(DIALOG_PATH_FILTER)!=null) {
-				String filter = settings.get(DIALOG_PATH_FILTER);
+			if (dialogSettings.getLastFilters() != null) {
+				String filter = dialogSettings.getLastFilters().getFirst();
+
+				searchIn.setItems(searchInStrings(dialogSettings.getLastFilters()));
```

Comment:
```suggestion
				searchIn.setItems(dialogSettings.getLastFilters().toArray(String[]::new));
```
and then you can remove the entire `searchInStrings()` method.

Reply:
done